packages/discord.js/src/util/EnumResolvers.js

```diff
@@ -0,0 +1,223 @@
+      case 'PRIMARY':
+        return ButtonStyle.Primary;
+      case 'SECONDARY':
+        return ButtonStyle.Secondary;
+      case 'SUCCESS':
+        return ButtonStyle.Success;
+      case 'DANGER':
+        return ButtonStyle.Danger;
+      case 'LINK':
+        return ButtonStyle.Link;
+      default:
+        return unknownKeyStrategy(key);
+    }
+  }
+
+  static resolveMessageType(key) {
+    switch (key) {
+      case 'DEFAULT':
+        return MessageType.Default;
+      case 'RECIPIENT_ADD':
+        return MessageType.RecipientAdd;
+      case 'RECIPIENT_REMOVE':
+        return MessageType.RecipientRemove;
+      case 'CALL':
+        return MessageType.Call;
+      case 'CHANNEL_NAME_CHANGE':
+        return MessageType.ChannelNameChange;
+      case 'CHANNEL_ICON_CHANGE':
+        return MessageType.ChannelIconChange;
+      case 'CHANNEL_PINNED_MESSAGE':
+        return MessageType.ChannelPinnedMessage;
+      case 'GUILD_MEMBER_JOIN':
+        return MessageType.GuildMemberJoin;
+      case 'USER_PREMIUM_GUILD_SUBSCRIPTION':
+        return MessageType.UserPremiumGuildSubscription;
+      case 'USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_1':
+        return MessageType.UserPremiumGuildSubscriptionTier1;
+      case 'USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_2':
+        return MessageType.UserPremiumGuildSubscriptionTier2;
+      case 'USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_3':
+        return MessageType.UserPremiumGuildSubscriptionTier3;
+      case 'CHANNEL_FOLLOW_ADD':
+        return MessageType.ChannelFollowAdd;
+      case 'GUILD_DISCOVERY_DISQUALIFIED':
+        return MessageType.GuildDiscoveryDisqualified;
+      case 'GUILD_DISCOVERY_REQUALIFIED':
+        return MessageType.GuildDiscoveryRequalified;
+      case 'GUILD_DISCOVERY_GRACE_PERIOD_INITIAL_WARNING':
+        return MessageType.GuildDiscoveryGracePeriodInitialWarning;
+      case 'GUILD_DISCOVERY_GRACE_PERIOD_FINAL_WARNING':
+        return MessageType.GuildDiscoveryGracePeriodFinalWarning;
+      case 'THREAD_CREATED':
+        return MessageType.ThreadCreated;
+      case 'REPLY':
+        return MessageType.Reply;
+      case 'CHAT_INPUT_COMMAND':
+        return MessageType.ChatInputCommand;
+      case 'THREAD_STARTER_MESSAGE':
+        return MessageType.ThreadStarterMessage;
+      case 'GUILD_INVITE_REMINDER':
+        return MessageType.GuildInviteReminder;
+      case 'CONTEXT_MENU_COMMAND':
+        return MessageType.ContextMenuCommand;
+      default:
+        return unknownKeyStrategy(key);
+    }
+  }
+}
+
+// Precondition logic wrapper
+function preconditioner(func) {
+  return key => {
+    if (typeof key !== 'string' && typeof key !== 'number') {
+      throw new Error('Enum value must be string or number');
+    }
+
+    if (typeof key === 'number') {
+      return key;
+    }
+
+    return func(key);
+  };
+}
+
+// Injects wrapper into class static methods.
+function applyPreconditioner(obj) {
+  Object.getOwnPropertyNames(obj).forEach(name => {
```

Comment:
I don’t recall seeing many uses of forEach in the library, wouldn’t it be better to use for of?